sympy/combinatorics/pc_groups.py

```diff
@@ -106,17 +110,18 @@ def minimal_uncollected_subwords(self, word):
             s2, e2 = array[i+1]
             s = ((s1, 1), )
             s = group.dtype(s)
-            if e1 > 0 and e1 < re[s]:
-                if e2 > 0 and index[s1] > index[s2]:
-                    # case-0:  v = x[i]**a*x[i+1], where index[x[i]] > index[x[i+1]]
-                    uncollected_subwords[((s1, e1), (s2, 1))] = 0
-
-                elif e2 < 0 and index[s1] > index[s2]:
-                    # case-1: v = x[i]**a*x[i+1]*-1, where index[x[i]] > index[x[i+1]]
-                    uncollected_subwords[((s1, e1), (s2, -1))] = 1
             if e1 > re[s]-1:
                 # case-2: v = x[i]**a
                 uncollected_subwords[((s1, e1), )] = 2
+                continue
```

Comment:
Why `continue`. It will suffice to return a single uncollected word. The other uncollected words (if there are any) may not be there after the one word has been handled.

Reply:
I thought we do not need to handle `x1**3` and `x1**3*x0`(cases like this) at the same time, once the power will be reduced then we can process that!

Reply:
I think that it is better to handle all uncollected minimal subwords as soon as we find them. They may cause extra work later if the power relation is more complicated than `x1**3 = 1`.